pandas/core/indexes/multi.py

```diff
@@ -734,21 +734,23 @@ def _values(self) -> np.ndarray:
                 vals = cast("CategoricalIndex", vals)
                 vals = vals._data._internal_get_values()
 
-            if isinstance(vals, ABCDatetimeIndex):
+            is_dti = isinstance(vals, ABCDatetimeIndex)
+
+            if is_dti:
```

Comment:
Did the same thing happen to `TimedeltaIndex`?

Reply:
No, the underlying TimeDelta objects don't have the freq, so as type(object) a couple of lines below this return the correct objects without resetting the freq